thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/completeness/checker/DataCompletenessScheduler.java

```diff
@@ -24,18 +24,14 @@
   private DataCompletenessJobRunner dataCompletenessJobRunner;
   private DataCompletenessJobContext dataCompletenessJobContext;
   private List<String> datasetsToCheck = new ArrayList<>();
-  private static final ThirdEyeCacheRegistry CACHE_REGISTRY = ThirdEyeCacheRegistry.getInstance();
 
   public DataCompletenessScheduler(String datasetsToCheck) {
     scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
     if (StringUtils.isNotBlank(datasetsToCheck)) {
       this.datasetsToCheck = Lists.newArrayList(datasetsToCheck.split(","));
-    } else {
-      this.datasetsToCheck = CACHE_REGISTRY.getCollectionsCache().getCollections();
```

Comment:
So far this part just NPEs later down the road. One approach I'd suggest is indexing over dataset requiring completeness checks and selecting them.
That design assumes that we consider "completeness" as dataset property rather than something associated with, for example, an anomaly algorithm. It's something we'll have to decide eventually.